### PR TITLE
chore: create base eslint config

### PR DESCRIPTION
**Pull request by araujogui**
## Description

Create base ESLint config

## Validation

Well, ESLint works

## Related Issues

Related to #5405

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: apps/site/eslint.config.js [hunk 1]
```diff
@@ -98,7 +48,7 @@ export default tseslint.config(
   {
     files: ['**/*.{mdx,tsx}'],
     rules: {
-      '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+      // '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
       'react/no-unescaped-entities': 'off',
```
Comment (araujogui) [inline apps/site/eslint.config.js]: I'm not sure how to enable this rule. When I try, I get the error "Key '@typescript-eslint/consistent-type-definitions': Could not find plugin '@typescript-eslint'," but isn't the plugin enabled in the base config?

File: package.json [hunk 2]
```diff
@@ -41,14 +41,20 @@
     "turbo": "2.1.2"
   },
   "devDependencies": {
+    "@eslint/js": "~9.10.0",
+    "@types/eslint__js": "~8.42.3",
     "commitizen": "4.3.0",
     "cz-conventional-changelog": "3.3.0",
+    "eslint": "~9.10.0",
+    "eslint-import-resolver-typescript": "~3.6.3",
```
Comment (ovflowd) [inline package.json]: This should not be on the base config.

File: package.json [hunk 3]
```diff
@@ -41,14 +41,20 @@
...
+    "eslint-plugin-import-x": "~4.2.1",
+    "eslint-plugin-no-relative-import-paths": "~1.5.5",
     "prettier": "3.3.3",
-    "prettier-plugin-tailwindcss": "0.6.6"
+    "prettier-plugin-tailwindcss": "0.6.6",
+    "typescript": "~5.5.4",
```
Comment (ovflowd) [inline package.json]: No TypeScript on base config 🙏

**Review by ovflowd — APPROVED** [review]
Appreciate the effort here! But the base config should have nothing TypeScript related, afaik!

File: apps/site/eslint.config.js (hunk 1 above)
Comment (ovflowd) [reply apps/site/eslint.config.js]: I answered on another comment!

**Review by aymen94 — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (araujogui) [reply package.json]: Isn't Typescript used in all our projects? Can you explain me why this is wrong?

Comment (ovflowd) [reply package.json]: Yes-ish, but in different ways. Each package has its own way of declaring TypeScript and its own rules.

For example, the i18n package has barely any TS besides the index.d.ts, and we don't compile TypeScript.

Comment (araujogui) [reply package.json]: got it, tks

**Comment by bmuenzenmeyer** [comment]
@araujogui do you have time rebase this? sorry it's been getting stale

**Comment by araujogui** [comment]
@nodejs/nodejs-website reviews pls

**Review by AugustinMauroy — APPROVED** [review]
LGTM !
As suggested in the turbo documentation, you can also use a package as a basis for the configuration
https://turbo.build/repo/docs/guides/tools/eslint#sharing-configuration

**Pull request merged**